fnv1/hash32.go

```diff
@@ -21,24 +21,20 @@ func HashUint32(u uint32) uint32 {
 
 // AddString32 adds the hash of s to the precomputed hash value h.
 func AddString32(h uint32, s string) uint32 {
-	i := 0
-	n := (len(s) / 8) * 8
-
-	for i != n {
-		h = (h * prime32) ^ uint32(s[i])
-		h = (h * prime32) ^ uint32(s[i+1])
-		h = (h * prime32) ^ uint32(s[i+2])
-		h = (h * prime32) ^ uint32(s[i+3])
-		h = (h * prime32) ^ uint32(s[i+4])
-		h = (h * prime32) ^ uint32(s[i+5])
-		h = (h * prime32) ^ uint32(s[i+6])
-		h = (h * prime32) ^ uint32(s[i+7])
-
-		i += 8
+	for len(s) >= 8 {
+		h = (h * prime32) ^ uint32(s[0])
+		h = (h * prime32) ^ uint32(s[1])
+		h = (h * prime32) ^ uint32(s[2])
+		h = (h * prime32) ^ uint32(s[3])
+		h = (h * prime32) ^ uint32(s[4])
+		h = (h * prime32) ^ uint32(s[5])
+		h = (h * prime32) ^ uint32(s[6])
+		h = (h * prime32) ^ uint32(s[7])
+		s = s[8:]
 	}
 
-	for _, c := range s[i:] {
-		h = (h * prime32) ^ uint32(c)
+	for i := range s {
```

Comment:
Any reason you're not doing the additional unrolling for `>=4` and `>=2` on this one?

Reply:
Just a miss, thanks for pointing it out!